packages/navie/src/commands/observe-command.ts

```diff
@@ -0,0 +1,143 @@
+import VectorTermsService from '../services/vector-terms-service';
+import { ContextV2 } from '../context';
+import { ExplainOptions } from './explain-command';
+import Message from '../message';
+
+const RelevantTest = z.object({
+  relevantTest: z
+    .object({
+      name: z.string().optional().describe('The name of the test case, if known'),
+      path: z.string().describe('The path to the test file'),
+      language: z
+        .enum(['ruby', 'python', 'java', 'javascript', 'other'])
+        .describe('The programming language of the test file'),
+      framework: z.string().optional().describe('The test framework used'),
+    })
+    .optional()
+    .describe('A descriptor of the most relevant test to the requested behavior'),
+});
+
+export default class ObserveCommand implements Command {
+  constructor(
+    private readonly options: ExplainOptions,
+    private readonly completionService: CompletionService,
+    private readonly lookupContextService: LookupContextService,
+    private readonly vectorTermsService: VectorTermsService
+  ) {}
+
+  getTestSnippets(vectorTerms: string[], tokenLimit: number): Promise<ContextV2.FileContextItem[]> {
+    return this.lookupContextService
+      .lookupContext(vectorTerms, tokenLimit, {
+        include: ['test', 'spec'],
+      })
+      .then((contextItems) =>
+        contextItems.filter(
+          (item): item is ContextV2.FileContextItem =>
+            item.type === ContextV2.ContextItemType.CodeSnippet
+        )
+      );
+  }
+
+  async getMostRelevantTest(
+    userRequest: string,
+    fileContextItems: ContextV2.FileContextItem[]
+  ): Promise<z.infer<typeof RelevantTest>['relevantTest']> {
+    const context = fileContextItems.map(
+      ({ content, location }) =>
+        `<code-snippet location="${location}">\n${content}\n</code-snippet>`
+    );
+    const prompt = `Given the following code snippets, identify the single most relevant test to the user request.
+<code-snippets>
+${context.join('\n')}
+</code-snippets>`;
+
+    const response = await this.completionService.json(
+      [
+        {
+          role: 'system',
+          content: prompt,
+        },
+        {
+          role: 'user',
+          content: userRequest,
+        },
+      ],
+      RelevantTest,
+      { model: ModelType.Mini }
+    );
+
+    return response?.relevantTest;
+  }
+
+  async *execute({ question: userRequest, userOptions }: CommandRequest): AsyncIterable<string> {
+    const vectorTerms = await this.vectorTermsService.suggestTerms(userRequest);
+    const tokenLimit = userOptions.numberValue('tokenlimit') || this.options.tokenLimit;
+    const testSnippets = await this.getTestSnippets(vectorTerms, tokenLimit);
+    const mostRelevantTest = await this.getMostRelevantTest(userRequest, testSnippets);
+    if (!mostRelevantTest) {
+      yield 'Sorry, I could not find any relevant tests to record.';
+      return;
+    }
+
+    if (mostRelevantTest.language === 'other') {
```

Comment:
If the user has appmap.yml we would want to direct the LLM to ONLY consider tests in the configured language.

Reply:
Specifying or overriding the language would also be a good thing to expose as an /option